examples/fibonacci.rs

```diff
@@ -109,6 +108,8 @@ fn fibo_circuit<F: Field + From<u64> + Hash>() -> (Circuit<F>, Option<TraceGener
             })
         });
 
+        ctx.pragma_num_steps(11);
+
```

Comment:
Probably should have mentioned in the last PR, but looks like `num_steps` in AST is used to calculate `num_rows` in CompilationUnit, which is used to calculate the number of assignment rows. This is different from `first_step` and `last_step`, which create a fixed column eventually.

My question is: How does `num_steps` enforce the number of steps if not through a fixed column? Can the user still pass in a witness with different number of steps while still passing the circuit constraints?

Reply:
https://github.com/privacy-scaling-explorations/chiquito/pull/58/files#diff-19a5e6a0406f9184fd74dba07991bee321e608450fff0e30117f02c49c97ea8aR585-R586

Reply:
Then I'd just comment that we are only allowing circuits with the same height for all steps, which I think we discussed last time and are fine with it for now.